node/src/service.rs

```diff
@@ -34,6 +29,45 @@ type FullClient = sc_service::TFullClient<Block, RuntimeApi, Executor>;
 type FullBackend = sc_service::TFullBackend<Block>;
 type FullSelectChain = sc_consensus::LongestChain<FullBackend, Block>;
 
+// TO DO inherent data provider
+pub const INHERENT_IDENTIFIER: sp_inherents::InherentIdentifier = *b"dhtdataM";
+
+impl ProvideInherentData for DataMap {
+    fn on_register(&self, _: &InherentDataProviders) -> Result<(), sp_inherents::Error> {
+		Ok(())
+	}
+
+    fn inherent_identifier(&self) -> &'static sp_inherents::InherentIdentifier {
+		// let key = format!("{:?}", self.key);
+		//Todo: convert keys above to bytes array. 
+		&INHERENT_IDENTIFIER
+    }
+
+    fn provide_inherent_data(&self, inherent_data: &mut sp_consensus::InherentData) -> Result<(), sp_inherents::Error> {
+        // dht value: block
+		// let mut data = InherentData::new();
+		inherent_data.put_data(INHERENT_IDENTIFIER, &self.value)
+		// todo!()
+    }
+
+    fn error_to_string(&self, error: &[u8]) -> Option<String> {
+        sp_inherents::Error::decode(&mut &error[..]).map(|e| e.into_string()).ok()
+    }
+}
+
+pub fn build_inherent_data_providers(key: String) -> Result<InherentDataProviders, ServiceError> {
+	let providers = InherentDataProviders::new();
+
+	let key_new= key.into_bytes();
```

Comment:
Can we avoid creating this variable and use `key.into_bytes()` inside `register_provider`?

Reply:
spot-on